prdoc/pr_8108.prdoc

```diff
@@ -0,0 +1,11 @@
+title: "[pallet_balances] Account for reserved balance in `ensure_can_withdraw` function"
+
+doc:
+- audience: [Runtime Dev, Runtime User]
```

Comment:
Consider splitting these out, as the runtime user will be most interested that it fixes the 
`LiquidityRestrictions` bug in production for proxies etc

Reply:
Improved the PR doc in https://github.com/paritytech/polkadot-sdk/pull/8108/commits/075cd9902fc9bca022a90d4b3453a39686614f31